### PR TITLE
Add ability to create polymorphic selects for polymorphic belongs to relationships.

### PR DESCRIPTION
**Pull request by brianjlandau**
- Create global_id instance methods for polymorphic relationships
- Add `polymorphic_select` method to the `FormBuilder`

### CODE REVIEW

File: spec/belongs_to_builder_extension_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe RailsPolymorphicSelect::BelongsToBuilderExtension do
+  let(:zebra){ create(:animal, name: 'Zebra', scientific_name: 'Equus zebra') }
+  let(:cheetah){ create(:animal, eats: zebra)}
+
+  it 'creates a global id reader method for polymorphic belongs to relationships' do
+    expect(Animal.public_method_defined?(:eats_global_id)). to eql(true)
```
Comment (rysospicy) [inline spec/belongs_to_builder_extension_spec.rb]: Got a space between `.` and `to`.

File: lib/rails_polymorphic_select/belongs_to_builder_extension.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    def define_global_id_methods(mixin, name)
+      mixin.class_eval <<-CODE, __FILE__, __LINE__ + 1
+        def #{name}_global_id
+          #{name}.try(:to_global_id)
+        end
+
+        def #{name}_global_id=(new_global_id)
+          self.#{name} = if new_global_id.present?
+            GlobalID::Locator.locate(new_global_id)
```
Comment (rysospicy) [inline lib/rails_polymorphic_select/belongs_to_builder_extension.rb:19]: Did not know about `GlobalID` in Rails... neat!

**Comment by rysospicy** [comment]
:+1:

File: lib/rails_polymorphic_select/belongs_to_builder_extension.rb (hunk 2 above)
Comment (brianjlandau) [inline lib/rails_polymorphic_select/belongs_to_builder_extension.rb:19]: Yeah! Got added in 4.2, I just learned about it.

**Comment by efatsi** [comment]
Neat stuff, excellent tests :+1:

**Comment by brianjlandau** [comment]
Thanks @h0tl33t & @efatsi!

**Pull request merged**